### ISSUE TITLE
Active Class

### ISSUE THREAD

**Issue by vilbht** [Shortrope/AccordionJS#1]
Hi,

Just found the plugin, very useful especially the flexibility not limited to a specific container. 
Had 3 queries:

1) How to add an expand collapse (+/-) and active CSS classes to the button triggers and to the content containers?

2) any tip to introduce functionality such that a Click outside accordion container (i.e. anywhere else in the body) collapses the accordion again.

3) In which line of the JS file should I declare singleOpen as true or false?

Thanks!

**Comment by Shortrope** [comment]
Hello Vilav,  and thanks for your input.  
1) You are right, it does need an expand/collapse (+/-)  
2) A click anywhere in the body closes open items...  You could add another click eventLisener on the body element and zero all the heights.   Something like:

``` js
document.querySelector('body').addEventListener('click', zeroAllHeights);
```

Not sure I will implement this but might create the zeroAllHeights() function for readability  
3) Where to declare 'singleOpen' true/false...  At the bottom of the file, in the call to addAccordionListener(..,..,.., true)
if true only one item can be open at a time.
if set to false, after an item is opened it will remain open until clicked again.  
I think I will rethink the addAccordionListener().  I don't think I need to pass in all those parameters  
Also need to implement when the document is ready so the script can be called in the head.  
I'll take care of these things this week.

**Comment by vilbht** [comment]
Thanks a lot @Shortrope! Will try out the suggested steps
Looking fwd to the updates to this plugin.

**Comment by Shortrope** [comment]
Made considerable changes to the code.  
Created two option vars: singleOpen and icons
Now using an object to store the target ids and associated data
Script can be placed in the head

**Comment by vilbht** [comment]
Thanks for the changes mate! The +/- works swell, especially the fact that the remote and regular +/- both toggle

Found a good reference to check out ( in the Nav) :
https://devart.withgoogle.com/
- The CSS styles of the trigger button change on hover / click
- Accordion collapses if clicked in any area below it or any link inside it.

Any tips on how I can incorporate the above two features into the plugin?

For the click anywhere to dismiss feature, I tried adding the document.querySelector('body').addEventListener('click', zeroAllHeights);
but couldn't get it to work

Thanks @Shortrope!
